### ISSUE TITLE
Invalid symmetric difference expression

### ISSUE THREAD

**Issue by domenkozar** [pre-commit/pre-commit#777]
```
$ git push
An unexpected error has occurred: CalledProcessError: Command: ('/nix/store/4c12r17gwha2hh6gvc6y2i0s6ckjc09y-git-2.16.2/libexec/git-core/git', 'diff', '--name-only', '--no-ext-diff', '-z', '192a93890ea5100a22a42a73249f7c1e571ebb0b...857ce8997c4f431c06ec52ef0ba1a64983ae945f')
Return code: 128
Expected return code: 0
Output: (none)
Errors:
    fatal: Invalid symmetric difference expression 192a93890ea5100a22a42a73249f7c1e571ebb0b...857ce8997c4f431c06ec52ef0ba1a64983ae945f


Check the log at /home/ielectric/.cache/pre-commit/pre-commit.log
```

I understand this happens if one rewrites git history, but pre-commit shouldn't fail in that case.

**Comment by asottile** [comment]
I agree! thanks for the report -- this is definitely a bug :)

**Comment by asottile** [comment]
I'm actually having a difficult time reproducing this, I've tried amending and pushing and the diff call succeeds for me -- could you help me reproduce this :)

**Comment by asottile** [comment]
actually no, I've got it now -- the remote progresses to a place that hasn't been fetched yet and then the push is attempted.

working on a fix now

**Comment by domenkozar** [comment]
Thank you!

**Comment by asottile** [comment]
This has been released as part of [v1.10.3](https://github.com/pre-commit/pre-commit/releases/tag/v1.10.3) -- thanks again for the issue 🎉
